### PR TITLE
[basic-sort] sorting added; package updates, cleanup

### PR DESCRIPTION
**Pull request by jtrein**
Recent updates to Blueprint Table and Core have brought improved table perf. I tested their old version with slightly more rows and columns and it was 🐢..

* Updates sort icon (no extraneous styles needed from Material UI) to be simple SVG
* Adds basic sorting: ASC, DESC
* Package cleanup from template (fetch, async, regenerator, etc.)
* Updates blueprint core and table, react bump

I briefly tested on local Universal Codes and it's 👌🏽 .

### CODE REVIEW

**Comment by jtrein** [comment]
FYI, want to use the updated table for mSupply Customer module, so getting this in now.

**Comment by jtrein** [comment]
Chris, if you have time this week, could you review this one? We need it for the msupply-customer module.

**Comment by Chris-Petty** [comment]
Whoa sorry. I was in TL when you made this, just back in when you last commented. Am back from surgery now.

File: src/SussolReactTable.jsx [hunk 1]
```diff
@@ -2,8 +2,25 @@ import React from 'react';
...
+* Sort example: 10 will come after 2, not "1, 10, 2"
+*
+* @param {any}     data which can be converted to a string
+* @param {any}     data which can be converted to a string
+* @param {bool}    isAscending: true/false
+* @return {int}    e.g. -1, 0, 1
+*/
+const compare = (a, b, isAscending) => (
+  isAscending
+  ? a.toString().localeCompare(b, undefined, { numeric: true, sensitivity: 'base' })
```
Comment (Chris-Petty) [inline src/SussolReactTable.jsx]: localeCompare built into Node? I gotta take note of this one

File: src/SussolReactTable.jsx [hunk 2]
```diff
@@ -42,24 +81,27 @@ export class SussolReactTable extends React.Component {
...
+      <ColumnHeaderCell>
         <FlatButton
           label={column.title}
           labelPosition="before"
           icon={sortIcon}
-          onClick={() => this.toggleSortOrder(column.key)}
+          onClick={
+            column.sortable
+            ? () => this.toggleSortOrder(column.key)
+            : noop
```
Comment (Chris-Petty) [inline src/SussolReactTable.jsx]: null?

**Comment by Chris-Petty** [comment]
yarn lock conflicts RIP

File: src/SussolReactTable.jsx (hunk 2 above)
Comment (jtrein) [reply src/SussolReactTable.jsx]: Probs wrote this too fast. `null` should work, I believe.

File: src/SussolReactTable.jsx (hunk 1 above)
Comment (jtrein) [reply src/SussolReactTable.jsx]: Those mental JS spec people!

File: src/SussolReactTable.jsx (hunk 2 above)
Comment (jtrein) [reply src/SussolReactTable.jsx]: Just checked. It's b/c Material UI `EnhancedButton` wants a func prop sent down, or it throws.

**Comment by jtrein** [comment]
Should be all set to go!

**Comment by jtrein** [comment]
* Moved deps on `material-ui` to `peerDependencies`. Otherwise, the bundle becomes bloated in projects where there is already `material-ui` present. The table assumes the dependent is already using a Material-like theme by including it as a design decision.
* Cleaned up some devDeps that are from the template we're not using
* Added a basic test, or two

**Pull request merged**